### ISSUE TITLE
[Bug]: NPU FlashAttention ignores causal=True in the dense attention path

### ISSUE THREAD

**Issue by Oliver7th** [vllm-project/vllm-omni#7322]
### Your current environment

  - Hardware: Ascend 950PR, single NPU
  - Container: `quay.io/ascend/vllm-omni:v0.28.0-a5`
  - Attention backend: `FLASH_ATTN`
  - Model: `nvidia/Cosmos3-Nano`

### Your code version

  - vLLM-Omni base: `v0.28.0-101-g087ac463`
    - Base commit: `087ac46318c42d1fa1e4de13a8053f94683898bd`
  - vLLM: `0.28.0+empty`
    - Source commit: `2cf0a6915ce544dc493a0990f2ea38d81601128a`
  - vLLM-Ascend: package metadata reports `0.0.0`
    - Source checkout is provided by the container image
    - Exact source commit is unavailable because the image does not retain its `.git` directory
  - PyTorch: `2.10.0+cpu`
  - torch-npu: `2.10.0.post4`
  - MindIE-SD: `3.1.0`
  - diffusers: `0.40.0`
  - transformers: `5.14.1`

### 🐛 Describe the bug

  On Ascend NPU, FlashAttentionImpl.forward_fa_npu() does not apply the
  causal constraint when the attention layer is constructed with
  causal=True.

  The CUDA FlashAttention paths forward self.causal to the FlashAttention
  kernel. However, the NPU dense path only forwards attn_mask to
  mindiesd.attention_forward():

  attention_mask = _maybe_reshape_attn_mask(
      query,
      key,
      attention_mask,
      mask_mode="full_qk",
  )

  return attention_forward(
      query,
      key,
      value,
      attn_mask=attention_mask,
      ...
  )

  When attn_metadata does not contain an explicit mask,
  attention_mask remains None, even if self.causal is True.
  The NPU attention is therefore executed as unmasked bidirectional
  attention.

  Cosmos3 exposes this problem because its understanding pathway constructs
  the self-attention layer with causal=True.

  ### Reproduction

  Start Cosmos3-Nano on one Ascend 950PR NPU:

  export ASCEND_RT_VISIBLE_DEVICES=<device-id>
  export VLLM_OMNI_LOGGING_LEVEL=DEBUG

  vllm serve <COSMOS3_NANO_PATH> \
    --served-model-name "nvidia/Cosmos3-Nano" \
    --omni \
    --host 0.0.0.0 \
    --port 8000 \
    --init-timeout 1800 \
    --model-class-name Cosmos3OmniDiffusersPipeline \
    --no-guardrails \
    --allowed-local-media-path / \
    --enforce-eager \
    --max-model-len 8192 \
    --tensor-parallel-size 1 \
    --vae-use-tiling \
    --diffusion-attention-backend FLASH_ATTN

  Send a text-to-video request:

  curl -X POST http://127.0.0.1:8000/v1/videos/sync \
    -H "Accept: video/mp4" \
    -F 'model=nvidia/Cosmos3-Nano' \
    -F 'prompt=A robot arm is cleaning a plate in the kitchen' \
    -F 'negative_prompt=blurry, distorted, low quality, jittery, deformed' \
    -F 'size=1280x720' \
    -F 'num_frames=25' \
    -F 'fps=24' \
    -F 'num_inference_steps=35' \
    -F 'guidance_scale=6.0' \
    -F 'max_sequence_length=4096' \
    -F 'flow_shift=10.0' \
    -F 'seed=123' \
    -o output.mp4

  ### Actual behavior

  The request completes and returns a video, but the generated frames are
  semantically meaningless noise.

  Instrumenting or mocking mindiesd.attention_forward() shows that
  attn_mask=None is passed even though the attention implementation has
  self.causal=True.

  No exception is raised, which makes this a silent correctness issue.

  ### Expected behavior

  The NPU backend should preserve the causal semantics requested by the model.

  For query length Sq and key length Skv, the mask should use the same
  bottom-right alignment as FlashAttention:

  key_position <= query_position + (Skv - Sq)

  If an explicit keep-mask is also supplied, the effective mask should be the
  logical intersection of the explicit mask and the causal mask.

  Non-causal attention should remain unchanged.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.
